e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -65,5 +69,112 @@ internal static class TestHelper {
+        /// </summary>       
+        /// <param name="deviceId">Password for HTTP basic authentication</param>
+        /// <param name="patch">Name of deployed Industrial IoT</param
+        private static async Task UpdateTagAsync(string patch) {
+            var deviceId = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_ID");
+        
+            Assert.True(!string.IsNullOrWhiteSpace(deviceId), "deviceId string is null");
+
+            var twin = await _registryManager.GetTwinAsync(deviceId);
+            await _registryManager.UpdateTwinAsync(twin.DeviceId, patch, twin.ETag);
+        }
+
+        /// <summary>
+        /// transfer the content of published_nodes.json file into the OPC Publisher edge module
+        /// </summary>       
+        /// <param name="sourceFilePath">Source file path</param>
+        /// <param name="destinationFilePath">Destination file path</param
+        public static void LoadPublishedNodesFile(string sourceFilePath, string destinationFilePath) {
+            var username = Environment.GetEnvironmentVariable("PCS_SIMULATION_USER");
+            var password = Environment.GetEnvironmentVariable("PCS_SIMULATION_PASSWORD");
+            var host = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_DNS_NAME");
+           
+            Assert.True(!string.IsNullOrWhiteSpace(username), "username string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(password), "password string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(host), "host string is null");
+
+            using (ScpClient client = new ScpClient(host, username, password)) {
+                client.Connect();
+
+                if (string.IsNullOrEmpty(sourceFilePath)) {
+                    DeletePublishedNodesFile(destinationFilePath);
+                }
+                using (Stream localFile = File.OpenRead(sourceFilePath)) {
+                    client.Upload(localFile, destinationFilePath);
+                }
+                client.Disconnect();
+            }
+        }
+
+        /// <summary>
+        /// Delete published_nodes.json file into the OPC Publisher edge module
+        /// </summary>       
+        /// <param name="sourceFilePath">Source file path</param>
+        /// <param name="destinationFilePath">Destination file path</param
+        public static bool DeletePublishedNodesFile(string destinationFilePath) {
+            var username = Environment.GetEnvironmentVariable("PCS_SIMULATION_USER");
+            var password = Environment.GetEnvironmentVariable("PCS_SIMULATION_PASSWORD");
+            var host = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_DNS_NAME");
+            
+            Assert.True(!string.IsNullOrWhiteSpace(username), "username string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(password), "password string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(host), "host string is null");
+
+            var isSuccessful = false;
```

Comment:
[suggestion] directly assert that isSuccessful is true, in test we try to fail early if something is wrong

Reply:
done